src-tools/netutils/asio_send_block/asio_send_block.cpp

```diff
@@ -46,40 +153,140 @@ class c_maintask {
+}
+
+void c_maintask::print_usage() const {
+		std::cout << std::endl;
 		std::cout << "Usage: ./client <host> <port> <max_speed> SendCommand" << std::endl;
-		std::cout << "e.g.:  ./client <host> <port> <max_speed> <msg> <msgbytes> <count> " << std::endl;
+		std::cout << "e.g.:  ./client <host> <port> <max_speed> <msg> <msgbytes> <count> [maxTime sec] [maxData GB]" << std::endl;
 		std::cout << "e.g.:  ./client 127.0.0.1 9000  999000    foo   1500       -1 " << std::endl;
+		std::cout << "e.g.:  ./client 127.0.0.1 9000  999000    foo   1500       -1 10 -1 " << std::endl;
+		std::cout << "\nor start server accepting remote RPC commands: \n";
+		std::cout << "e.g.:  ./client remote_cmd 192.168.70.17 9000 " << std::endl;
+		std::cout << "or instead allow remote access (remote controll of this sending, WARNING can make your computer spam/DDoS other computer) " << std::endl;
+		std::cout << "e.g.:  ./client remote <listen_on_ip><port>  <authorized_ip> <max_time_hours> <pass>  " << std::endl;
+		std::cout << "e.g.:  ./client remote 192.168.70.17 9011   192.168.70.16   72 secret1" << std::endl;
+		std::cout << "e.g.:  ./client remote 192.168.70.17 9011   0.0.0.0         4  secret1" << std::endl;
+		std::cout << "e.g.:  ./client remote       0.0.0.0 9011   0.0.0.0         4  secret1" << std::endl;
+		std::cout << "then send TCP text like e.g.: \n";
+		print_usage_rpc_cmd();
+		std::cout << std::endl;
+}
+
+std::string c_maintask::run_rpc_command_string(const std::string & rpc_cmd) {
+	std::string ret;
+	ret += "RPC-result ";
+	std::cout << "RPC command to execute: [" << rpc_cmd << "]\n" << std::endl;
+	std::vector<std::string> args = explode( remove_extra_whitespace(rpc_cmd) , ' ');
+	std::cout << "RPC command args size: " << args.size() << std::endl;
+
+	try {
+		std::string password_given = args.at(0);
+		std::string word_SEND = args.at(1);
+
+		if (m_rpc_password.size()<3) throw std::runtime_error("insecure (short) or missing RPC password (configured here)");
+		if (password_given != m_rpc_password) throw std::runtime_error("invalid RPC password given");
+		if (word_SEND != "SEND") throw std::runtime_error("word 'SEND'");
+
+		args.erase( args.begin() , args.begin()+2 );
+
+	} catch(const std::exception &ex) {
+		_warn("Can not parse RPC: " << ex.what() << " rpc was [" << rpc_cmd << "]");
+		throw ;
+	}
+
+	args.insert( args.begin(), "program_name" );
+
+	size_t argc = args.size();
+	const char ** argv = new const char * [ argc ];
+	for (size_t i=0; i<argc; ++i) argv[i] = args.at(i).c_str(); // ! points to memory owned by strings in vector args!
+	for (size_t i=0; i<argc; ++i) _info("argv["<<i<<"] = [" << argv[i] << "]");
+	std::string error_msg="(no error)";
+	try {
+
+		c_maintask maintask_once; // fresh new program (to not reuse our object this)
+		maintask_once.run( argc, argv );
+
+	} catch (const std::exception & ex) {
+		error_msg = ex.what();
+		_warn("Exception while tryint to execute RPC command: " << error_msg);
+	}
+	delete [] argv;
```

Comment:
memleak if strange exception was thrown

Reply:
not possible, but used vector so no problem now